ast/misc.go

```diff
@@ -387,10 +387,8 @@ func (n *ExecuteStmt) Accept(v Visitor) (Node, bool) {
 // See https://dev.mysql.com/doc/refman/5.7/en/commit.html
 type BeginStmt struct {
 	stmtNode
-	Mode                  string
-	ReadOnly              bool
-	Bound                 *TimestampBound
-	CausalConsistencyOnly bool
```

Comment:
Is this filed do not used?

Reply:
https://github.com/pingcap/parser/pull/1162, looks like no.

Reply:
After checking the TiDB code, I think this is used. Maybe I shouldn't remove it.

Reply:
https://docs.pingcap.com/tidb/stable/transaction-overview#causal-consistency

Reply:
@nolouch @youjiali1995 I added it back and added more test cases. PTAL.